tests/Type/Nette/data/json_decode.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+use Nette\Utils\Json;
+use function PHPStan\Testing\assertType;
+
+$value = Json::decode('true');
+assertType('bool', $value);
+
+$value = Json::decode('1');
+assertType('int', $value);
+
+$value = Json::decode('1.5');
+assertType('float', $value);
+
+$value = Json::decode('false');
+assertType('bool', $value);
```

Comment:
Could even be constant-false?

Reply:
I'll try to use the `resolveConstantStringType()`, we'll see :)